docs/adr/0010-migrate-to-v3-cluster-resource.md

```diff
@@ -0,0 +1,68 @@
+- In the case of the current controller, we generate a v1 and let Rancher take care of v3.
+- When migrating to the newer controller, we generate a v3 and let Rancher take care of v1.
+
+#### So, what is the difference between both controllers?
+
+Refer to the [original v3 controller ADR](./0008-managementv3-clusters-support.md) if you want to dive into details of how the import controller works. The following is just a summary for contextualizing this ADR specifically.
+
+The main difference in terms of logic is that, while watching v1 resources, we can use the name of the clusters to identify whether there is an existing resource that represents the CAPI-provisioned cluster.
+
+This means that the resource itself is easily linked to a CAPI cluster via name.
+
+Alternatively, when watching v3 clusters in the new controller, the name of the resource is a semi-random string automatically generated by Kubernetes, which forces us to find a different approach to find the resource of interest. Additionally, v3 clusters are not namespaced but global.
+To simplify the watcher on v3 clusters, we add two labels to the resource to reference the `name` and `namespace` of the CAPI cluster it is linked to.
+
+#### And, what is the difference between v3 clusters for each controller?
+
+Now that we now how we link Rancher cluster -> CAPI cluster, we know that the difference between v3 clusters in both controllers is the existence of the CAPI cluster `name` and `namespace` labels.
+
+This means that, if the new controller we'll be migrating to is able to identify v3 clusters with the required labels, we can simply switch controllers and expect everything to be unchanged and Rancher clusters to maintain the same configuration.
+
+#### How does this affect import controllers?
+
+With the current import mechanism, Rancher automatically creates a v3 clusters that links to the v1 cluster generated by Turtles. As stated before, this means that no new resources need to be created for the new import controller to operate normally.
+During migration Turtles needs to be aware of the existing imported clusters and apply labels accordingly so these resources can be reused. As part of the migration guide, we will provide a migration script that allows users to prepare imported clusters for migration in batches, which will simplify the migration process (you can refer to [this PR](https://github.com/rancher/turtles/pull/574)). During this process, there needs to be a mechanism to identify which clusters are actually managed by Turtles: this can simply be done by looking for the `-capi` suffix in the resource name, which is added to all Rancher cluster resources created from Turtles. Additionally, after resources have been prepared for migration, they must be marked to avoid repeating the process on the same resource more than once. An annotation on the `clusters.provisioning.cattle.io` will be used for this:
```

Comment:
Should this also mention that we are matching the clusters using `cluster-api.cattle.io/owned` label under the hood in our controller? It feels like undocumented detail in the ADRs flow, and it takes part in applying `migrated` annotation.